File: src/main/java/com/phong/taskmanagement/entity/Project.java
package com.phong.taskmanagement.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String projectCode;

    @Column(columnDefinition = "NVARCHAR(255)")
    private String projectName;

    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String description;

    private LocalDate startDate;

    private LocalDate endDate;

    private String status;

    @OneToMany(mappedBy = "project")
    @JsonIgnore
    private List<Task> tasks = new ArrayList<>();
}

File: taskmanagement-ui/src/types/project.ts
export interface Project {
  id: number;
  projectCode: string;
  projectName: string;
  description: string;
  startDate: string; // ISO format YYYY-MM-DD
  endDate: string | null;
  status: string;
}

export interface ProjectSearchParams {
  keyword?: string;
  status?: string;
  page?: number;
  size?: number;
}

export interface CreateProjectPayload {
  projectCode: string;
  projectName: string;
  description: string;
  startDate: string;
  endDate?: string | null;
  status: string;
}

export interface UpdateProjectPayload extends CreateProjectPayload {}
